src/project/types/website/website-navigation-md.ts

```diff
@@ -192,6 +196,39 @@ const prevPageTitleHandler = (context: NavigationPipelineContext) => {
   };
 };
 
+const safeKey = (key: string) => {
```

Comment:
@dragonstyle I'm not sure if refactoring to a Pure Function with unit tests would make sense here, but I thought I would suggest it.  There is growing complexity in this area.

Reply:
This overall change looks a lot more complicated than it really is. The root issue here was my decision to try to reuse the rendered sidebar to compute the proper rendered contents of the breadcrumbs by just plucking the contents of the sidebar from the DOM. This is ok in normal cases, but in some cases (for example, sections in the sidebar which are just text with no href), it breaks down. Rather than doing the "clever" thing, this ultimately just does what we do pretty much everywhere else- just render the contents of the breadcrumbs with our markdown pipeline. Most of the diff (especially in website navigation is just removing a few lines of code and moving code).

The growing complexity I think reflected a couple failed attempts to fix the issue with band aids and finally this, the actual fix :)